salesagility/SuiteCRM
# Fix #10150 - Allow internal enpoints in iframe dashlet

## Description

Created a new function getCustomEntrypoints() to find all the custom entry points, then modified the function isSelfRequest($endpoint) to check the list of custom entry points in which case it will return false and allow the entry point.


1. Create a custom entry point.
2. Add a web iframe dashlet to your dashboard with the interal URL
3. It will not be allowed.
Apply the fix, and it will be allowed.

https://github.com/salesagility/SuiteCRM/issues/10150


## Motivation and Context

This allows internal entrypoints to be allowed to be embedded in a dashlet.  One would presume that internal entry points created by a developer should be allowed to be embeded.

## How To Test This


Apply the patch and then create a custom entry point.  Like echo "hello world"; and then try and embed that URL in Dashlet as iframe.  It should be allowed with these modificaitons.

## Types of changes

- [ x] Bug fix (non-breaking change which fixes an issue)

### Final checklist


- [ x] My code follows the code style of this project found [here](https://docs.suitecrm.com/community/contributing-code/coding-standards/).
- [ x] I have read the [**How to Contribute**](https://docs.suitecrm.com/community/contributing-code/) guidelines.
